update to version 2.0.0

Murmele opened the pull request:
fixes: #51

Murmele committed:
update to version 2.0.0

Murmele commented:
bot, ping admins

finish-args-host-tmp-access: Is required to share files to other applications
finish-args-has-socket-ssh-auth: Required to use ssh connection for git
finish-args-home-filesystem-access: Required to clone and accessing repositories in the home directory
finish-args-flatpak-spawn-access: Required to execute another application like (KDiff3 for external merges)

hfiguiere commented on com.github.Murmele.Gittyup.yml:
```diff
@@ -1,6 +1,6 @@
 app-id: com.github.Murmele.Gittyup
 runtime: org.kde.Platform
-runtime-version: 5.15-23.08
+runtime-version: 6.9
```
current is 6.10

Murmele replied:
```suggestion
runtime-version: 6.10
```

hfiguiere commented:
these are not the build error...

Murmele committed:
update to 6.10

hfiguiere commented:
> finish-args-host-tmp-access: Is required to share files to other applications finish-args-has-socket-ssh-auth: Required to use ssh connection for git finish-args-home-filesystem-access: Required to clone and accessing repositories in the home directory finish-args-flatpak-spawn-access: Required to execute another application like (KDiff3 for external merges)


These are not the reason the build fails. Read the message. The appstream file is invalid.

Murmele commented:
It looks like below. Why the text is a problem?
When looking on older version for me it looks fine: https://flathub.org/de/apps/com.github.Murmele.Gittyup
```xml
<?xml version="1.0" encoding="UTF-8"?>
<component type="desktop-application">
  <id>com.github.Murmele.Gittyup</id>
  <metadata_license>FSFAP</metadata_license>
  <project_license>MIT</project_license>
  <name>Gittyup</name>
  <summary>Graphical Git client designed to help you understand and manage your source code history</summary>
  <developer_name>Gittyup Community</developer_name>

  <description>
    <p>Graphical Git client designed to help you understand and manage your source code history</p>

    <p>Features:</p>
    <ul>
		<li>Fast native interface designed to help you understand and manage your source code history</li>
        <li>Staging of single lines</li>
        <li>Double tree view: Seeing staged and unstaged changes in different trees.</li>
        <li>Maximize History or Diff view by pressing Ctrl+M</li>
        <li>Ignore Pattern: Ability to ignore all files defined by a pattern instead of only one file</li>
        <li>Tag Viewer: When creating a new tag all available tags are visible. Makes it easier to create consistent tags.</li>
        <li>Commit Message template: Making it easier to write template based commit messages.</li>
    </ul>
  </description>

  <launchable type="desktop-id">com.github.Murmele.Gittyup.desktop</launchable>

  <screenshots>
    <screenshot type="default">
      <caption>The main view</caption>
        <image>https://raw.githubusercontent.com/Murmele/Gittyup/master/rsrc/screenshots/main_dark_orig.png</image>
    </screenshot>
    <screenshot>
      <caption>Staging single lines</caption>
        <image>https://raw.githubusercontent.com/Murmele/Gittyup/master/rsrc/screenshots/double_treeview_single_line_staging.png</image>
    </screenshot>
    <screenshot>
      <caption>Tree View</caption>
        <image>https://raw.githubusercontent.com/Murmele/Gittyup/master/rsrc/screenshots/treeview.png</image>
    </screenshot>
  </screenshots>

  <url type="homepage">https://murmele.github.io/Gittyup</url>
  <url type="vcs-browser">https://github.com/Murmele/Gittyup</url>
  <url type="bugtracker">https://github.com/Murmele/Gittyup/issues</url>
  <url type="help">https://matrix.to/#/#Gittyup:matrix.org</url>
  <url type="donation">https://liberapay.com/Gittyup/donate</url>

  <provides>
    <binary>gittyup</binary>
  </provides>

  <releases>
	<release version='v2.0.0' date='2025-11-30'>
	<description>
	Bug Fix and Feature release
	<p>Added</p>
	<ul>
	<li>Add commit filter to show only the first parent in the commit list view</li>
	<li>Add backend support for SHA256 repos</li>
	<li>Fix email filter being not returning any results</li>
	<li>Add mmap based reading to speed up indexer</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Use Qt6</li>
	<li>Update libgit2 library</li>
	</ul>
	</description>
	</release>
	<release version='v1.4.0' date='2024-04-24'>
	<description>
	Bug Fix and Feature release
	<p>Added</p>
	<ul>
	<li>UI(Commit List): Added a right-click menu entry to rename branches.</li>
	<li>UI(Main Menu): Added a menu-entry to rename the current branch.</li>
	<li>UI(Diff View): Added line wrapping option in Tools - Options - Diff</li>
	<li>Option to open Gittyup maximized at startup</li>
	<li>Hotkeys for Navigating CommitList</li>
	<li>Referring repos with their git-dir instead of their work-tree.</li>
	<li>Use columns for file name, directory, and state when files are shown as a list in TreeViews.</li>
	<li>Possibility to change language from the settings</li>
	<li>Linux: Publishing Appimage bundle</li>
	<li>Hide Untracked Files option to DoubleTreeWidget</li>
	<li>Add display options to the commit list</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>UI(Commit List): Collapse multiple branch and tag right-click menu entries
	into submenus. This affects the checkout and delete operations.</li>
	<li>Fix(Build System): Force usage of clang-format v13 to ensure consistent formatting.</li>
	<li>Fix: Commit highlighting</li>
	<li>Compact commit list: move commit message closer to commit graph</li>
	<li>Updating translations</li>
	<li>Close the branch drop-down list after the user checks out a branch.</li>
	<li>Fix: Delete timeline causes crash on windows</li>
	<li>Fix: Rebase: Use original commit message as default message</li>
	<li>Fix: Stage all button</li>
	</ul>
	</description>
	</release>
	<release version='v1.3.0' date='2023-04-20'>
	<description>
	Performance Improvement and feature release
	<p>Added</p>
	<ul>
	<li>Colorized status badges</li>
	<li>Template: use first template as default template for the commit message</li>
	<li>Search function for the treeview</li>
	<li>Reworked credential store: add possibility to choose between different methods to store credentials</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Fix external diff in Flatpak build</li>
	<li>Fix windows credentials</li>
	<li>Fix force push to correct remote</li>
	<li>Fix tab title if more than three times a repository with the same name is opened</li>
	<li>Fix storing repository settings correctly, because otherwise they are not applied</li>
	<li>Fix language support</li>
	<li>Improved refresh velocity</li>
	<li>Fix storing and restoring current opened file when Gittyup refreshes</li>
	<li>Improved velocity for files with many hunks</li>
	</ul>
	</description>
	</release>
	<release version='v1.2.2' date='2023-01-22'>
	<description>
	Bug fix release
	<p>Changed</p>
	<ul>
	<li>Fix flatpak install process</li>
	</ul>
	</description>
	</release>
	<release version='v1.2.1' date='2023-01-22'>
	<description>
	Bug fix release
	<p>Added</p>
	<ul>
	<li>Possibility to hide avatar (Settings - Window - View - Show Avatars)</li>
	<li>Show log entry when a conflict during rebase happens</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Fix download url for flatpak and macos</li>
	<li>Fix Segmentation fault when ignoring files</li>
	<li>Fix discard of complete files and submodules</li>
	<li>Fix context menu entries</li>
	<li>Fix bytesize overflow</li>
	<li>Fix focus loose during scrolling in the Commitlist with the keyboard</li>
	<li>Do not crash when the repository is for some reason broken</li>
	<li>Fix crash if rebasing is not possible</li>
	</ul>
	</description>
	</release>
	<release version='v1.2.0' date='2022-10-28'>
	<description>
	Bug fix and feature release
	<p>Added</p>
	<ul>
	<li>Add support for solving merge conflicts for whole files</li>
	<li>Solving binary conflicts directly in Gittyup</li>
	<li>Support rebasing with conflict solving</li>
	<li>Implement amending commits</li>
	<li>Possibility to init submodules after clone (Settings - General - Update submodules after pull and clone)</li>
	<li>Hiding menu bar (Application Settings - Window - Hide Menubar)</li>
	<li>Implement support for Gitea instances</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Fix Segmentation fault when using space to stage files</li>
	<li>Fix menubar color in dark theme</li>
	<li>Filter only branches, tags, remotes attached to selected commit</li>
	<li>Fix crash when global GIT config is invalid</li>
	<li>Fix crash when having errors while adding a remote account</li>
	<li>Fix updater on windows, macos and linux (flatpak)</li>
	<li>Fix discarding file leading to discarding submodule changes</li>
	<li>Fix rebase log messages during rebase</li>
	<li>Improve SSH config handling</li>
	<li>Application settings and repository settings can now be selected with a single settings button</li>
	<li>Use the full file context menu for the staging file list</li>
	<li>Fix Arch Linux build</li>
	</ul>
	</description>
	</release>
	<release version='v1.1.2' date='2022-08-12'>
	<description>
	Bug fix release
	<p>Changed</p>
	<ul>
	<li>Fix bundled OpenSSL version incompatibility</li>
	</ul>
	</description>
	</release>
	<release version='v1.1.1' date='2022-06-09'>
	<description>
	Bug fix release
	<p>Added</p>
	<ul>
	<li>Distinguish between commit author and committer</li>
	<li>Show image preview also for deleted files</li>
	<li>Official macOS release</li>
	<li>Show which kind of merge conflict occurred for each conflict</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Fix single line staging if not all hunks are loaded</li>
	<li>Fix cherrypick commit author</li>
	<li>Fix segmentation fault if submodule update fails</li>
	<li>Fix line staging with windows new lines</li>
	<li>Show first change in the diff view when loading</li>
	<li>Improved windows icon</li>
	</ul>
	</description>
	</release>
	<release version='v1.1.0' date='2022-04-30'>
	<description>
	Second release of Gittyup
	<p>Added</p>
	<ul>
	<li>Button to directly access the terminal and the filebrowser</li>
	<li>Add support for running in single instance mode</li>
	<li>Customizable hotkeys</li>
	<li>Quick commit author overriding</li>
	<li>keyboard-interactive SSH auth</li>
	<li>Improved single line staging and replacing staging image to a more appropriate one</li>
	<li>Font customizing</li>
	<li>Options to switch between staging/unstaging treeview, single tree view and list view</li>
	<li>Do not automatically abort rebase if conflicts occur</li>
	<li>Add possibility to save file of any version on local system</li>
	<li>Add possibility to open a file of any version with default editor</li>
	</ul>
	</description>
	</release>
	<release version='v1.0.0' date='2021-11-18'>
	<description>
	First version of the GitAhead Fork Gittyup
	<p>Added</p>
	<ul>
	<li>Staging of single lines</li>
	<li>Double tree view: Seeing staged and unstaged changes in different trees.</li>
	<li>Maximize History or Diff view by pressing Ctrl+M</li>
	<li>Ignore Pattern: Ability to ignore all files defined by a pattern instead of only one file</li>
	<li>Tag Viewer: When creating a new tag all available tags are visible. Makes it easier to create consistent tags.</li>
	<li>Commit Message template: Making it easier to write template based commit messages.</li>
	</ul>
	</description>
	</release>
	
  </releases>

 <content_rating type="oars-1.1">
    <content_attribute id="violence-cartoon">none</content_attribute>
    <content_attribute id="violence-fantasy">none</content_attribute>
    <content_attribute id="violence-realistic">none</content_attribute>
    <content_attribute id="violence-bloodshed">none</content_attribute>
    <content_attribute id="violence-sexual">none</content_attribute>
    <content_attribute id="violence-desecration">none</content_attribute>
    <content_attribute id="violence-slavery">none</content_attribute>
    <content_attribute id="violence-worship">none</content_attribute>
    <content_attribute id="drugs-alcohol">none</content_attribute>
    <content_attribute id="drugs-narcotics">none</content_attribute>
    <content_attribute id="drugs-tobacco">none</content_attribute>
    <content_attribute id="sex-nudity">none</content_attribute>
    <content_attribute id="sex-themes">none</content_attribute>
    <content_attribute id="sex-homosexuality">none</content_attribute>
    <content_attribute id="sex-prostitution">none</content_attribute>
    <content_attribute id="sex-adultery">none</content_attribute>
    <content_attribute id="sex-appearance">none</content_attribute>
    <content_attribute id="language-profanity">none</content_attribute>
    <content_attribute id="language-humor">none</content_attribute>
    <content_attribute id="language-discrimination">none</content_attribute>
    <content_attribute id="social-chat">none</content_attribute>
    <content_attribute id="social-info">none</content_attribute>
    <content_attribute id="social-audio">none</content_attribute>
    <content_attribute id="social-location">none</content_attribute>
    <content_attribute id="social-contacts">none</content_attribute>
    <content_attribute id="money-purchasing">none</content_attribute>
    <content_attribute id="money-gambling">none</content_attribute>
  </content_rating>
  <keywords>
    <keyword>git client</keyword>
    <keyword>source code manager</keyword>
    <keyword>version control tool</keyword>
  </keywords>

</component>
```

hfiguiere commented:
the validation didn't check that before. it does now. It was always invalid.

Murmele commented:
This is fine for me, but I don't really understand what talks against this structure we currently have. Because rendered on Flathub it looks fine for me.

barthalion commented:
It's unhappy about some text in release description not being in the `<p>` tags.

While we already patch out libappstream, you will need to take your grievances upstream. We don't control all of this.

Murmele commented:
> It's unhappy about some text in release description not being in the `<p>` tags.
> 
> While we already patch out libappstream, you will need to take your grievances upstream. We don't control all of this.

So you mean that it is not the `<p>Added</p> but the line above?
```
<description>
	Bug fix release
	<p>Added</p>
	<ul>
	<li>Possibility to hide avatar (Settings - Window - View - Show Avatars)</li>
	</ul>
	<p>Changed</p>
	<ul>
	<li>Fix download url for flatpak and macos</li>
	</ul>
</description>
```

barthalion commented:
Yeah, that's my guess.

Murmele commented:
Currently we are using the following command in the CI to validate the file:

`flatpak run --env=G_DEBUG=fatal-criticals org.freedesktop.appstream-glib validate <path>/com.github.Murmele.Gittyup.appdata.xml`

Using `appstream.cli` returns the errors here so I can reproduce

Murmele committed:
fix metainfo

Murmele committed:
Merge remote-tracking branch 'refs/remotes/origin/v2.0.0'
